snuba/query/processors/tagsmap.py

```diff
@@ -103,6 +104,14 @@ def __is_optimizable(
             )
         return None
 
+    def __find_tags(self, expression: Expression) -> bool:
+        for node in expression:
+            if isinstance(node, Column) and node.column_name.startswith(
+                self.__nested_col
+            ):
```

Comment:
`startswith` seems like the wrong method to use here, since it could result in other columns that just incidentally have the same prefix to return `True` here. Isn't there a better option? I assume this has to match both `col.xyz` and `col[key]` forms (and possibly `col` for ARRAY JOIN?)

Reply:
I think it is right. I will change it.
The columns it needs to match are "tags[aaa]" as well as tags_key and tags_value. But, thinking over it, the risk of not de-optimizing the tags for tags_key is lower than matching a column that just happens to start with `tag`